### PR TITLE
Standalone folowup tests

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Add more tests verifying the following conditions:
- discovery of DI providers from exported NgModules
- forwardRef in standalone component imports

### CODE REVIEW

File: packages/core/src/render3/jit/directive.ts [hunk 1]
```diff
@@ -210,12 +210,17 @@ function getStandaloneDependencies(imports: Type<any>[]): R3TemplateDependencyFa
...
     if (anyDef) {
       if (!anyDef.standalone) {
-        // TODO: real error
-        throw new Error(`What are you doing? You imported a non-standalone thing!`);
+        throw new Error(`The imported ${
+            compDef    ? 'component' :
+                dirDef ? 'directive' :
+                         'pipe'} "${
+            stringifyForError(
+                dep)}" is not standalone, did you forget add the standalone: true flag?`);
```
Comment (JoostK) [inline packages/core/src/render3/jit/directive.ts:223]: `forget add` -> `forget to add`

File: packages/core/src/render3/jit/directive.ts [hunk 2]
```diff
@@ -210,12 +210,17 @@ function getStandaloneDependencies(imports: Type<any>[]): R3TemplateDependencyFa
...
-    const dirDef = getComponentDef(dep) || getDirectiveDef(dep);
-    const anyDef = dirDef || getPipeDef(dep);
+    const compDef = getComponentDef(dep);
+    const dirDef = getDirectiveDef(dep);
+    const anyDef = compDef || dirDef || getPipeDef(dep);
     if (anyDef) {
       if (!anyDef.standalone) {
-        // TODO: real error
-        throw new Error(`What are you doing? You imported a non-standalone thing!`);
+        throw new Error(`The imported ${
```
Comment (JoostK) [inline packages/core/src/render3/jit/directive.ts:218]: Can we use `RuntimeError` here with a dedicated error code?

File: packages/core/src/render3/jit/directive.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply packages/core/src/render3/jit/directive.ts]: Good catch, thnx!

File: packages/core/src/render3/jit/directive.ts (hunk 2 above)
Comment (pkozlowski-opensource) [reply packages/core/src/render3/jit/directive.ts]: Probably! Not super-familiar with JiT error reporting and saw that we are just  throwing `Error` in other places (ex. module verification). But I would definitively want to follow the "best in class" error reporting here. TBD.

**Review by JoostK — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit 39c16818956baa49d66a190b40102cf554b4d887.

**Pull request closed without merging**